src/rgw/rgw_cache.cc

```diff
@@ -338,9 +338,34 @@ void ObjectCache::unchain_cache(RGWChainedCache *cache) {
 }
 
 ObjectCache::~ObjectCache()
-{
+{ 
+  cct->_conf.remove_observer(this);
   for (auto cache : chained_cache) {
     cache->unregistered();
   }
 }
 
+const char** ObjectCache::get_tracked_conf_keys() const
+{
+  static const char* keys[] = {
+    "rgw_cache_lru_size",
+    "rgw_cache_expiry_interval",
+    nullptr
+  };
+  return keys;
+}
+
+ void ObjectCache::handle_conf_change(const ConfigProxy& conf,
+           const std::set<std::string>& changed)
+{
+  if (changed.find("rgw_cache_lru_size") != changed.end()) {
+    RWLock::WLocker l(lock);
+    lru_max_size = cct->_conf.get_val<int64_t>("rgw_cache_lru_size");
+    lru_window = lru_max_size / 2;
```

Comment:
if we're reducing the value of `lru_max_size` here, we need to invalidate any entries above that size. i don't see any other logic in ObjectCache that would do that for us